File: checkout-service/src/foam.kt
package com.affirm.checkout

import com.foam.kotlin.Foam
import com.foam.kotlin.FoamConfig

fun initializeFoam() {
    Foam.init(
        FoamConfig(
            serviceName = "checkout-service",
            apiKey = System.getenv("FOAM_API_KEY"),
            isProduction = System.getenv("ENVIRONMENT") == "production",
            enableTraces = true
        )
    )
}

File: loan-lifecycle-service/src/foam.py
import foam
from os import getenv

foam.init(
    service_name="loan-lifecycle-service",
    api_key=getenv("FOAM_API_KEY"),
    is_production=getenv("ENV") == "production",
    enable_traces=True,
    enable_metrics=True,
    enable_logs=True
)